File: security/security-oauth2/src/main/java/net/yan100/compose/security/oauth2/properties/WechatProperties.java
package net.yan100.compose.security.oauth2.properties;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.boot.context.properties.NestedConfigurationProperty;

@Data
@ConfigurationProperties(prefix = "compose.security.oauth2.wechat")
public class WechatProperties {
  @NestedConfigurationProperty
  private WxpaProperties wxpa = new WxpaProperties();
}

File: security/oauth2/src/main/kotlin/net/yan100/compose/security/oauth2/properties/WechatProperties.kt
package net.yan100.compose.security.oauth2.properties

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.NestedConfigurationProperty

private const val PREFIX = "compose.security.oauth2.wechat"

@ConfigurationProperties(prefix = PREFIX) data class WechatProperties(@NestedConfigurationProperty var wxpa: WxpaProperties = WxpaProperties())
